### PR TITLE
Openstack default fix

### PR DESCRIPTION
**Pull request by SimonRichardson**
## Description of change

The following also checks for empty default string, so that we're
backwards compatible with other providers like openstack. Openstack
here adds a default of "" and if a provider sets the default to ""
we skip it, like it was equal to nil.

Also updated to include drive by comment removal and better default
wording for LXD provider.

## QA steps

```sh
juju add-cloud
Cloud Types
  lxd
  maas
  manual
  oci
  openstack
  oracle
  vsphere

Select cloud type: openstack

Enter a name for your openstack cloud: os1 

Enter the API endpoint url for the cloud []: https://os1.com 

Enter a path to the CA certificate for your cloud if one is required to access it. (optional) [none]: 

Auth Types
  access-key
  userpass

Select one or more auth types separated by commas: userpass

Enter region name: 

Enter region name: default

Enter the API endpoint url for the region [use cloud api url]: https://region.os1.com

Enter another region? (y/N): n

Cloud "os1" successfully added
You may need to `juju add-credential os1' if your cloud needs additional credentials
Then you can bootstrap with 'juju bootstrap os1'
```

## Documentation changes

N/A

## Bug reference

N/A

### CODE REVIEW

**Comment by SimonRichardson** [comment]
https://trello.com/c/9bMocRmW/449-lxd-default-schema-region

File: cmd/juju/interact/pollster.go [hunk 1]
```diff
@@ -405,11 +405,21 @@ func (p *Pollster) queryAdditionalProps(vals map[string]interface{}, schema *jso
...
 		var name string
 		var err error
-		if schema.Default != nil {
+		if schema.Default != nil && schema.Default != "" {
 			var def string
 			if schema.PromptDefault != nil {
 				def = fmt.Sprintf("%v", schema.PromptDefault)
-			} else {
+			}
+			if len(schema.EnvVars) > 0 {
```
[the remark this answers is not in the document]
Comment (SimonRichardson) [reply cmd/juju/interact/pollster.go]: Agreed.

**Comment by SimonRichardson** [comment]
!!build!!

File: cmd/juju/interact/pollster_test.go [hunk 2]
```diff
@@ -681,3 +681,168 @@ Enter region name:
...
+		"name": "Bill",
+		"nested": map[string]interface{}{
+			"default": map[string]interface{}{
+				"name": "",
+			},
+		},
+	})
+}
+
+func (PollsterSuite) TestQueryObjectSchemaWithDefaultEnvVars(c *gc.C) {
```
Comment (hmlanigan) [inline cmd/juju/interact/pollster_test.go]: add TestQueryObjectSchemaEnvVarsWithOutDefault?

File: cmd/juju/interact/pollster.go [hunk 3]
```diff
@@ -405,26 +405,44 @@ func (p *Pollster) queryAdditionalProps(vals map[string]interface{}, schema *jso
 		// (i.e. map key) is the "name" of the thing.
 		var name string
 		var err error
-		if schema.Default != nil {
+		// Note: here we check that schema.Default is empty as well.
+		if schema.Default == nil || schema.Default == "" {
```
Comment (hmlanigan) [inline cmd/juju/interact/pollster.go]: Still failing :-( with OS_REGION_NAME set.  perhaps check to see that no schema.EnvVars available too?

File: cmd/juju/interact/pollster_test.go (hunk 2 above)
Comment (SimonRichardson) [reply cmd/juju/interact/pollster_test.go]: Done

**Comment by SimonRichardson** [comment]
$$merge$$

**Pull request merged**